### PR TITLE
Fix interpreting of ref.as_func|data

### PR DESCRIPTION
**Pull request by kripken**
(oops, how'd we miss that..?)

### CODE REVIEW

File: test/passes/Oz_fuzz-exec_all-features.wast [hunk 1]
```diff
@@ -325,6 +325,42 @@
...
+  (drop
+   ;; This should trap.
+   (ref.as_data
+    (ref.func $0)
+   )
+  )
+ )
+ (func "ref-as-data-of-data"
+  (drop
+   ;; This should trap.
```
Comment (aheejin) [inline test/passes/Oz_fuzz-exec_all-features.wast:338]: Why?

File: test/passes/Oz_fuzz-exec_all-features.wast [hunk 2]
```diff
@@ -325,6 +325,42 @@
...
+   (ref.as_func
+    (struct.new_default_with_rtt $struct
+     (rtt.canon $struct)
+    )
+   )
+  )
+ )
+ (func "ref-as-func-of-func"
+  (drop
+   ;; This should trap.
```
Comment (aheejin) [inline test/passes/Oz_fuzz-exec_all-features.wast:358]: Why? (2)

**Review by aheejin — APPROVED** [review]

**Review by tlively — APPROVED** [review]
LGTM when those comments @aheejin pointed out are removed or fixed.

File: test/passes/Oz_fuzz-exec_all-features.wast (hunk 1 above)
Comment (kripken) [reply test/passes/Oz_fuzz-exec_all-features.wast:338]: Oh, that looks like a bad copy-paste, sorry...

**Pull request merged**